app/src/plugins/sd-ct-oauth-plugin/services/passport.service.js

```diff
@@ -4,14 +4,53 @@ const { BasicStrategy } = require('passport-http');
 const TwitterStrategy = require('passport-twitter').Strategy;
 const FacebookStrategy = require('passport-facebook').Strategy;
 const GoogleStrategy = require('passport-google-oauth20').Strategy;
+const AppleStrategy = require('passport-apple');
 const LocalStrategy = require('passport-local').Strategy;
 const GoogleTokenStrategy = require('passport-google-token').Strategy;
 const FacebookTokenStrategy = require('passport-facebook-token');
 const bcrypt = require('bcrypt');
 const UserModel = require('plugins/sd-ct-oauth-plugin/models/user.model');
 
-
 function passportService(plugin) {
+    async function registerAppleUser(req, accessToken, refreshToken, decodedIdToken, profile, done) {
+        logger.info('[passportService - registerAppleUser] Registering user', profile);
+        logger.debug('[passportService - registerAppleUser] accessToken', accessToken);
+        logger.debug('[passportService - registerAppleUser] refreshToken', refreshToken);
+        logger.debug('[passportService - registerAppleUser] decodedIdToken', decodedIdToken);
+
+        let user = await UserModel.findOne({
+            provider: 'apple',
+            providerId: decodedIdToken.sub,
+        }).exec();
+        logger.info(user);
+        const { email } = decodedIdToken;
+        if (!user) {
+            logger.info('[passportService] User does not exist');
+            user = await new UserModel({
+                email,
+                provider: 'apple',
+                providerId: decodedIdToken.sub
+            }).save();
+        } else {
+            logger.info('[passportService] Updating email');
+            user.email = email;
+            await user.save();
+        }
+        logger.info('[passportService] Returning user');
+        done(null, {
+            // eslint-disable-next-line no-underscore-dangle
+            id: user._id,
+            provider: user.provider,
+            providerId: user.providerId,
+            role: user.role,
+            createdAt: user.createdAt,
+            extraUserData: user.extraUserData,
+            name: user.name,
+            photo: user.photo,
+            email: user.email
+        });
```

Comment:
:thinking: ?

```suggestion
        done(null, user.toJSON());
```

Reply:
my approach is more explicit as to what gets serialized and what doesn't. Typically this is preferable, as it requires explicit control over what gets passed out - for example, if you add a new field to the model, you need to explicitly add it here, and by default its not exposed. So if that field happens to be sensitive, by default, it stays hidden.

In the particular context of a user, this is even more important, as just blindly serializing data could expose passwords and salts, and that's halfway into getting attacked.